src/posix/fs.cpp

```diff
@@ -1203,24 +1215,23 @@ void PosixFileSystemAccess::emptydirlocal(string* name, dev_t basedev)
                  || *d->d_name != '.'
                  || (d->d_name[1] && (d->d_name[1] != '.' || d->d_name[2])))
                 {
-                    t = name->size();
-                    name->append("/");
-                    name->append(d->d_name);
+                    ScopedLengthRestore restore(name);
+                    PosixFileSystemAccess pfsa;
```

Comment:
creating and destroying this might have some performance impact: we could consider having a singleton instance used here and in `::dnext` too

Reply:
(thinking specially in inotify_init1 call)

Reply:
Yes good point.  I had checked the one for windows, which is trivial, but not this one.   In this case, we are actually in a member function of one of those, so we can just use its own localseparator :-)   adjusted.

Reply:
Ah, it turns out to be a static function, so I've put that back in, but outside the loop.  It's a function to delete a directory tree, so it won't be called particularly often, so one construction should be fine.  thanks